moveit_experimental/moveit_jog_arm/src/jog_interface_base.cpp

```diff
@@ -187,6 +187,22 @@ void JogInterfaceBase::jointsCB(const sensor_msgs::JointStateConstPtr& msg)
   shared_variables_mutex_.unlock();
 }
 
+bool JogInterfaceBase::changeControlDimensions(moveit_msgs::ChangeControlDimensions::Request& req,
+                                              moveit_msgs::ChangeControlDimensions::Response& res)
+{
+  shared_variables_mutex_.lock();
+  shared_variables_.control_dimensions[0] = req.control_x_translation;
+  shared_variables_.control_dimensions[1] = req.control_y_translation;
+  shared_variables_.control_dimensions[2] = req.control_z_translation;
+  shared_variables_.control_dimensions[3] = req.control_x_rotation;
+  shared_variables_.control_dimensions[4] = req.control_y_rotation;
+  shared_variables_.control_dimensions[5] = req.control_z_rotation;
+  shared_variables_mutex_.unlock();
```

Comment:
shouldn't there be more uses of the mutex? Where do the vars get read?

Reply:
These variables are read in `jog_calcs`, with mutex protection, [here](https://github.com/UTNuclearRoboticsPublic/moveit/blob/a39180947f56098c9fef9c8a42b60eb7ec809cec/moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp#L278-L285) (`src/jog_calcs.cpp` line 278)

Reply:
Ah, I thought that was some sort of reset function, I should have read more into it. Sorry.

Reply:
since the size is always 6, you could use a std::array<std::atomic<bool>, 6> for this and you would not need the mutex.

Reply:
nice!